### PR TITLE
Add Signal tests

### PR DESCRIPTION
**Pull request by Funth0mas**
I added some Signal tests and fixed two docstrings (problems found when writing tests).

What is your opiniton on TODO question in test_signal_set_startbit_raise()?

### CODE REVIEW

File: src/canmatrix/tests/test_canmatrix.py [hunk 1]
```diff
@@ -145,6 +147,138 @@ def test_ecu_repr():
...
+# Signal (generic functions)
+def test_signal_has_comment(some_signal):
+    comment = "comment01"
+    some_signal.addComment(comment)
+    assert some_signal.comment == comment
+
+
+def test_signal_find_attribute(some_signal):
+    some_signal.addAttribute("attr1", 255)
+    assert some_signal.attribute("attr1") == 255  # check value from 'attributes'
```
Comment (altendky) [inline src/canmatrix/tests/test_canmatrix.py]: Do these comments explain something the code doesn't?  Ditto elsewhere.

File: src/canmatrix/tests/test_canmatrix.py [hunk 2]
```diff
@@ -145,6 +147,138 @@ def test_ecu_repr():
     assert str(ecu) == "BoardUnit(name='Gateway', comment='with bug')"
 
 
+# Signal (generic functions)
+def test_signal_has_comment(some_signal):
+    comment = "comment01"
+    some_signal.addComment(comment)
+    assert some_signal.comment == comment
```
[the remark this answers is not in the document]
Comment (Funth0mas) [reply src/canmatrix/tests/test_canmatrix.py]: Delete, or let it be and only take the note in account?

File: src/canmatrix/tests/test_canmatrix.py (hunk 1 above)
Comment (Funth0mas) [reply src/canmatrix/tests/test_canmatrix.py]: I think that looking on test code it isn't obvious why testing "attr1" and "is_float" and also another values in the next TC. I wanted to state that one value is from class attribute, another from "user defined" `attributes` dict et so on. Maybe not well formulated.

File: src/canmatrix/tests/test_canmatrix.py [hunk 3]
```diff
@@ -3,6 +3,8 @@
 
 import canmatrix.canmatrix
 
+make_decimal = decimal.Decimal
```
[the remark this answers is not in the document]
Comment (altendky) [reply src/canmatrix/tests/test_canmatrix.py]: Side note:
I was thinking we should parameterize these against different float options. But I wasn't all that worried about it. Basically you'd have a test to verify the default is respected and everything else would be tested against both decimal and float and...  I dunno, fractions?  Honestly, each time the topic comes up I'm resisting being opinionated and just forcing decimal on everyone.  :|. I mean, it's not my decision and I don't really want to, but I do think about it.

On topic:
If this sticks around right now I'd use a name that matches that used elsewhere for the concept.  I might even just import and use the actual default.

File: src/canmatrix/tests/test_canmatrix.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (altendky) [reply src/canmatrix/tests/test_canmatrix.py]: Leave the test.  'side note' was meant to indicate the comment wasn't meant to affect this PR.

File: src/canmatrix/tests/test_canmatrix.py [hunk 4]
```diff
@@ -145,6 +147,138 @@ def test_ecu_repr():
...
+def test_signal_has_comment(some_signal):
+    comment = "comment01"
+    some_signal.addComment(comment)
+    assert some_signal.comment == comment
+
+
+def test_signal_find_attribute(some_signal):
+    some_signal.addAttribute("attr1", 255)
+    assert some_signal.attribute("attr1") == 255  # check value from 'attributes'
+    assert some_signal.attribute("is_float") is False  # check value from member attribute
```
[the remark this answers is not in the document]
Comment (Funth0mas) [reply src/canmatrix/tests/test_canmatrix.py]: I'm expecting exactly False, not something falsy... still I should set the is_float=False and don't rely on the default value, right?

File: src/canmatrix/tests/test_canmatrix.py (hunk 1 above)
Comment (altendky) [reply src/canmatrix/tests/test_canmatrix.py]: Yeah, I didn't get that, sorry.  I'd say that's two tests.  One to read 'builtin' attributes or whatever, and another for 'custom' or 'user'.  Whatever terminology is used elsewhere (if it's clear).  Then it moves what you wanted to express via the comments into code (and even test results).

File: src/canmatrix/tests/test_canmatrix.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (altendky) [reply src/canmatrix/tests/test_canmatrix.py]: Now that I get what this is doing, compare the function call to the direct attribute access. That's what you want to match.  My comments about `is` above still apply, so go ahead and use it.

File: src/canmatrix/tests/test_canmatrix.py [hunk 5]
```diff
@@ -145,6 +147,138 @@ def test_ecu_repr():
...
+def test_signal_set_startbit_conversion():
+    signal = canmatrix.canmatrix.Signal(size=16, is_little_endian=False)
+    signal.setStartbit(20, startLittle=True)
+    assert signal.startBit == 5  # lsb on 20, msb is 20-15 = 5
+    # TODO add test for reversed endianning
+
+
+def test_signal_set_startbit_raise():
+    signal = canmatrix.canmatrix.Signal(size=16, is_little_endian=False)
+    with pytest.raises(Exception):  # TODO change to ValueError?
```
Comment (altendky) [inline src/canmatrix/tests/test_canmatrix.py]: I'm still in favor of custom exceptions to isolate from stdlib exceptions. But yes, this is wildly vague.  Separate PR for just that (unless there are other exceptions to tidy similarly).

File: src/canmatrix/tests/test_canmatrix.py (hunk 1 above)
Comment (Funth0mas) [reply src/canmatrix/tests/test_canmatrix.py]: test splitted

File: src/canmatrix/tests/test_canmatrix.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Funth0mas) [reply src/canmatrix/tests/test_canmatrix.py]: staying with explicit Decimal.

**Pull request merged**